scripts/core/editor3/reducers/suggestions.jsx

```diff
@@ -77,24 +71,77 @@ const createDeleteSuggestion = (state, {data}) => {
     const selectionLength = selection.getEndOffset() - selection.getStartOffset();
 
     if (selectionLength === 0) {
-        editorState = setDeleteSuggestionForCharacter(editorState, data);
-    } else {
+        editorState = changeEditorSelection(editorState, -1, 0, false);
+    }
+    editorState = deleteCurrentSelection(editorState, data);
+
+    return onChange(state, editorState, true);
+};
+
+/**
+ * @ngdoc method
+ * @name pasteAddSuggestion
+ * @param {Object} state
+ * @param {Object} content - the suggestion added content
+ * @param {Object} data - info about the author of suggestion
+ * @return {Object} returns new state
+ * @description Add a new suggestion of type ADD based on content.
+ */
+const pasteAddSuggestion = (state, {content, data}) => {
+    const acceptedStyles = ['BOLD', 'ITALIC', 'UNDERLINE'];
+    let {editorState} = state;
+    const blockMap = content.getBlockMap();
+
+    editorState = deleteCurrentSelection(editorState, data);
+
+    blockMap.forEach((block) => {
+        if (block.getType() !== 'atomic') {
+            const text = block.getText();
+
+            for (let i = 0; i < text.length; i++) {
```

Comment:
I think you could use here `block.characterList.forEach` and then you can check the styles using [CharacterMetadata.hasStyle](https://draftjs.org/docs/api-reference-character-metadata.html#hasstyle)